### PR TITLE
Introduce MiniqmcOptions to capture miniQMC input options

### PR DESCRIPTION
**Pull request by PDoakORNL**

### CODE REVIEW

**Comment by ye-luo** [comment]
I had some headache using IBM XL with C++14. Keeping C++11 in miniQMC is appreciated.
But if you indeed use some C++14 feature, we should enforce C++14. There is no doubt.

File: src/Drivers/MiniQMCOptions.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        break;
+      case 'v':
+        mq_opt.verbose = true;
+        break;
+      case 'V':
+        ::print_version(true);
+        return mq_opt;
+        //should throw
+        break;
+      case 'w': // number of crowds
```
Comment (ye-luo) [inline src/Drivers/MiniQMCOptions.cpp]: Please keep the meaning of -w unchanged. -w is the number of walkers in the whole population within 1 MPI rank.
The number of crowds is more of an implementation detail and can be derived based on the population size and crowd size.

Comment (PDoakORNL) [reply src/Drivers/MiniQMCOptions.cpp]: I will leave it and add a different parameter for walkers per crowd.

**Comment by PDoakORNL** [comment]
c++14 is required for the parallel-block abstraction.

File: src/Drivers/MiniQMCOptions.cpp (hunk 1 above)
Comment (ye-luo) [reply src/Drivers/MiniQMCOptions.cpp]: This should work. Could you correct the comment next to case 'w' as well?

**Comment by ye-luo** [comment]
@PDoakORNL Could you name the C++14 feature in use?

**Comment by PDoakORNL** [comment]
I'm not sure of the exact "feature name" But I was needing changes in variadic template parameter pack expansion.  However checking to see exactly what it was it seems no longer present in this PR.

**Review by ye-luo — APPROVED** [review]

**Pull request merged**